faceted_search/src/index.js

```diff
@@ -1,65 +1,137 @@
+		let sortFields = encode(JSON.stringify(store.getState()['query']['sortFields']));
+		let searchParamsDict = {"searchFields": searchFields, "start": start, "sortFields": sortFields};
+		
+		// Update the query parameters with the latest values selected in the UI
+        setSearchParams(searchParamsDict);
+    },[searchParams, store.getState()]);
+    
+    return (
+        <div>
+            <SolrFacetedSearch
+                        {...store.getState()}
+                        {...solrClient.getHandlers()}
+                        bootstrapCss={true}
+                        onSelectDoc={(doc) => {console.log(doc); }} />
+        </div>
+    );
+}
+
 // Register your app with the store
 store.subscribe(() =>
-	// In stead of using the handlers passed along in the onChange callback of SolrClient
-	// use the .getHandlers() method to get the default click / change handlers
-	ReactDOM.render(
-
-		<SolrFacetedSearch
-			{...store.getState()}
-			{...solrClient.getHandlers()}
-			bootstrapCss={true}
-			onSelectDoc={(doc) => console.log(doc)}
-		/>,
-		document.getElementById("app")
-	)
+	// The inclusion of the BrowserRouter and Routes wrapping our APP is what allows the searchParams functionality to work.
+    ReactDOM.render(
+        <BrowserRouter>
+            <Routes>
+                <Route path="/" element={<APP />}/>
+            </Routes>
+        </BrowserRouter>
+        ,
+        document.getElementById("app")
+    )
 );
 
 document.addEventListener("DOMContentLoaded", () => {
-	// this will send an initial search initializing the app
-	
-	solrClient.initialize();
+    // this will send an initial search initializing the app
+	// We just need to set state when we firstly open the page with url
+	// So, we only need to set the initalize solrClient rather than set them in the useEffect
+	// Get the parameters when the page loads. 
+	// let [searchParams, setSearchParams] = useSearchParams();
+	// console.log(searchParams)
+	let CurURL = window.location.href;
+	let url = new URL(CurURL);
+	// Read the encoded fields out of the dictionary.  Note that these *must* match up with what we're encoding down above
+	let start = url.searchParams.get('start');
+	let searchFields = url.searchParams.get('searchFields');
+	let sortFields = url.searchParams.get('sortFields');
+
+	if (start && searchFields){
+		let decodedStart = JSON.parse(decode(start));
+		let decodedSearchFields = JSON.parse(decode(searchFields));
+		let decodedSortFields = JSON.parse(decode(sortFields));
+		// Update solrClient and request a new solr query
+		let ParamesDict = {'searchFields': decodedSearchFields, 'sortFields': decodedSortFields};
+
+		// Use solrClient built-in functions
+		// set initial query. This function would not send a query.
+		solrClient.setInitialQuery(ParamesDict);
+		// set page. This function will send a query.
+		solrClient.setCurrentPage(decodedStart)
```

Comment:
Use the built-in functions to update the solrClient parameters. `setInitialQuery` function only updates the searchFields and sortFields without sending a solr query. `setCurrentPage` will set a new page number and send a solr query.